.gitlab-ci.yml

```diff
@@ -162,10 +158,62 @@ check-rustdoc:
     RUSTDOCFLAGS:                  "-Dwarnings"
   script:
     - time cargo +nightly doc --workspace --all-features --verbose --no-deps
-    - sccache -s
+
+cargo-check-benches:
+  stage:                           test
+  <<:                              *docker-env
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
+  needs:
+    - job:                         check-rustdoc
+      artifacts:                   false
+  script:
+    - time cargo check --all --benches
+
+#### stage:                        build
+
+build-linux-stable:
+  stage:                           build
+  <<:                              *docker-env
+  <<:                              *collect-artifacts-short
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
+  needs:
+    - job:                         check-rustdoc
+      artifacts:                   false
+  script:
+    - mkdir -p ./artifacts
+    - time cargo build --release --locked --bin polkadot-parachain
```

Comment:
@PierreBesson please have a look.
We're already building a binary [up there](https://github.com/paritytech/cumulus/pull/1311/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R120). It goes with the debug assertions though. If you do not care, we could reuse it.